p2p/src/block_sync_service.rs

```diff
@@ -555,6 +559,65 @@ impl<P: Preset> BlockSyncService<P> {
 
                             self.request_blobs_and_blocks_if_ready();
                         }
+                        P2pToSync::RequestedExecutionPayloadEnvelope(envelope, peer_id, request_id, request_type) => {
+                            let block_root = envelope.message.beacon_block_root;
+
+                            self.sync_manager.record_received_execution_payload_envelope_response(
+                                block_root,
+                                peer_id,
+                                request_id,
+                            );
+
+                            let request_direction = match request_type {
+                                RPCRequestType::Root => SyncDirection::Forward,
+                                RPCRequestType::Range => self
+                                    .sync_manager
+                                    .request_direction(request_id)
+                                    .unwrap_or(self.sync_direction),
+                            };
+
+                            match request_direction {
+                                SyncDirection::Forward => {
+                                    let envelope_slot = envelope.message.slot;
+
+                                    if self.controller.contains_block(block_root)
+                                        && self.register_new_received_envelope(block_root, envelope_slot)
```

Comment:
```suggestion
                                    if self.register_new_received_envelope(block_root, envelope_slot)
```
whether the block is available or not, we just pass it to fork choice to decide
https://github.com/ethereum/consensus-specs/blob/11e15092dccdffed0db15a0314682ee0c5737eec/specs/gloas/p2p-interface.md?plain=1#L226-L228